handlers/handler.go

```diff
@@ -77,19 +78,46 @@ func IndexS3Object(s3objectURL string) {
 	log.Printf("Finish to compute hashes for %s", key)
 
 	indexdInfo, _ := getIndexServiceInfo()
-	rev, err := GetIndexdRecordRev(uuid, indexdInfo.URL)
-	if err != nil {
-		log.Println(err)
-		return
+
+	var retries = 0
+	var rev = ""
+
+	for {
+		rev, err = GetIndexdRecordRev(uuid, indexdInfo.URL)
+		if err != nil {
+			retries++
+			time.Sleep(30)
+		}
+		if retries == MaxRetries {
+			log.Println(err)
+			return
+		}
 	}
 
 	body := fmt.Sprintf(`{"size": %d, "urls": ["%s"], "hashes": {"md5": "%s", "sha1":"%s", "sha256": "%s", "sha512": "%s", "crc": "%s"}}`,
 		objectSize, s3objectURL, hashes.Md5, hashes.Sha1, hashes.Sha256, hashes.Sha512, hashes.Crc32c)
-	resp, err := UpdateIndexdRecord(uuid, rev, indexdInfo, []byte(body))
-	if err != nil {
-		log.Println(err)
-	}
 
-	log.Printf("Finish updating the record. Response Status: %s", resp.Status)
+	retries = 0
+	for {
+		resp, err := UpdateIndexdRecord(uuid, rev, indexdInfo, []byte(body))
+		if err != nil {
+			retries++
+			time.Sleep(30)
```

Comment:
Can you add logs here about the error and that we are trying + the attempt number?